wagtail/admin/tests/test_widgets.py

```diff
@@ -552,6 +554,39 @@ def test_tags_help_text_no_spaces_allowed(self):
             html,
         )
 
+    @override_settings(WAGTAIL_TAG_SPACES_ALLOWED=False, WAGTAIL_TAG_LIMIT=3)
+    def test_render_js_init_with_legacy_settings(self):
+        """Checks that the 'w-tag' controller attributes are correctly added to the tag widgets using legacy TAG_* settings"""
+        widget = widgets.AdminTagWidget()
+
+        html = widget.render("tags", None, attrs={"id": "alpha"})
+        params = self.get_js_init_params(html)
+
+        self.assertEqual(
+            params,
+            [
+                "alpha",
+                "/admin/tag-autocomplete/",
+                {"allowSpaces": False, "tagLimit": 3, "autocompleteOnly": False},
+            ],
+        )
+
+    @override_settings(TAG_SPACES_ALLOWED=False)
+    def test_tag_spaces_allowed_deprecation_warning(self):
+        """Checks that a deprecation warning is raised for TAG_SPACES_ALLOWED"""
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+
+            widget = widgets.AdminTagWidget()
+            widget.render("tags", None, attrs={"id": "alpha"})
+
+            self.assertTrue(
+                any(
+                    issubclass(warning.category, RemovedInWagtail70Warning)
+                    for warning in w
+                )
+            )
```

Comment:
This can be simplified with `self.assertWarns()`. We also want to assert the warning message – we can use `self.assertWarnsMessage()` for this. And the deprecation is only raised when `render()` is called, so let's minimise the scope of the context manager's block to be more specific. Finally, I think we should check that the setting is still reflected to the JS init params:

```suggestion
    @override_settings(TAG_SPACES_ALLOWED=False)
    def test_legacy_tag_spaces_allowed_setting(self):
        widget = widgets.AdminTagWidget()
        with self.assertWarnsMessage(
            RemovedInWagtail70Warning,
            "The setting 'TAG_SPACES_ALLOWED' is deprecated. "
            "Please use 'WAGTAIL_TAG_SPACES_ALLOWED' instead."
        ):
            widget.render("tags", None, attrs={"id": "alpha"})
        params = self.get_js_init_params(html)
        self.assertEqual(
            params,
            [
                "alpha",
                "/admin/tag-autocomplete/",
                {"allowSpaces": False, "tagLimit": None, "autocompleteOnly": False},
            ],
        )
```